substrate/pkg/controller/substrate/cluster/config.go

```diff
@@ -207,10 +210,27 @@ func (c *Config) ensureBucket(ctx context.Context, substrate *v1alpha1.Substrate
 	} else {
 		logging.FromContext(ctx).Infof("Created s3 bucket %s", aws.StringValue(discovery.Name(substrate)))
 	}
+	//sets tags on a bucket. Any existing tags are replaced.
+	if _, err := c.S3.PutBucketTagging(&s3.PutBucketTaggingInput{
+		Bucket: discovery.Name(substrate),
+		Tagging: &s3.Tagging{TagSet: []*s3.Tag{{
+			Key:   aws.String("kit.aws/substrate"),
+			Value: aws.String(substrate.Name)},
+		}},
+	}); err != nil {
+		return fmt.Errorf("adding tag %w", err)
+	}
```

Comment:
This is an old commit, its already merged in. You will need to rebase your repo.